src/crypto/sha512.cpp

```diff
@@ -205,3 +212,43 @@ CSHA512& CSHA512::Reset()
     sha512::Initialize(s);
     return *this;
 }
+
+int main(){
+
+
+    //Prepare test data
+    unsigned char buf[64] = {0};
+    buf[0] = 0xff;
+    buf[1] = 0xff;
+    buf[2] = 0xff;
+    uint64_t      *prt;
+    prt = (uint64_t*) buf;
+
+    //Initialize digest and process buffer
+    CSHA512 hasher;
+    hasher.Write(buf, 3);   //Tried both 3 and 64
```

Comment:
Err... if it's number of bytes maybe it needs to be 4?

Reply:
With 4, I get 

```
OpenSSL Sha512
Digest is: 62080346799e74025e65e5d6da46e4612303fda014c38a15cadf59251a1e0259f236ad265dfde3e26bf69970c901d652dc300d77b3179881e6dd410cc00fc487
 
Dogecoin's Bitcoin Sha512 Implementation (Input = 0xffffff )
Digest is: c154b71d83465bbfff995e8d3735b52c5c2db291cfbc32a49f1b60d74ebb1e0743243a9118ac37b2469aa5cd41836ce32e4f0ad93e165580f213823849515746
 
```

Reply:
It would be 3, because the line terminator does not count towards the hash, it does not get included. I also tried 64. Its still wrong.

Reply:
Yeah sorry just realised it's intentionally only using 3 bytes.